File: app/src/javax/java/com/adawolfs/mvp_java/login/dagger/LoginComponent.java
package com.adawolfs.mvp_java.login.dagger;

import com.adawolfs.mvp_java.login.LoginActivity;
import com.adawolfs.mvp_java.login.dagger.*;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by adawolfs on 1/20/17.
 */
@Singleton
@Component(modules = com.adawolfs.mvp_java.login.dagger.LoginModule.class)
public interface LoginComponent {
    void inject(LoginActivity __);
}

File: app/src/kotlin/java/com/adawolfs/mvp_java/login/dagger/LoginComponent.kt
package com.adawolfs.mvp_java.login.dagger

import com.adawolfs.mvp_java.login.LoginActivity
import com.adawolfs.mvp_java.login.dagger.*

import javax.inject.Singleton

import dagger.Component

/**
 * Created by adawolfs on 1/20/17.
 */
@Singleton
@Component(modules = arrayOf(com.adawolfs.mvp_java.login.dagger.LoginModule::class))
interface LoginComponent {
    fun inject(activity: LoginActivity)
}